ui/app/components/ttl-picker2.js

```diff
@@ -0,0 +1,105 @@
+/**
+ * @module TtlPicker2
+ * TtlPicker2 components are used to enable and select TTL
+ *
+ * @example
+ * ```js
+ * <TtlPicker2 @requiredParam={requiredParam} @optionalParam={optionalParam} @param1={{param1}}/>
+ * ```
+ * @param {function} onChange - This function will be passed a TTL object, which includes enabled{bool}, seconds{number}, timeString{string}.
+ * @param {string} [label='Time to live (TTL)'] - Label is the main label that lives next to the toggle.
+ * @param {string} [helperTextDisabled='Allow tokens to be used indefinitely'] - This helper text is shown under the label when the toggle is switched off
+ * @param {string} [helperTextEnabled='Disable the use of the token after'] - This helper text is shown under the label when the toggle is switched on
+ * @param {number} [time=30] - This is the time (in the default units) which will be adjustable by the user of the form
+ * @param {string} [unit='s'] - This is the unit key which will show by default on the form. Can be one of `s` (seconds), `m` (minutes), `h` (hours), `d` (days)
+ */
+
+import Ember from 'ember';
+import Component from '@ember/component';
+import { computed } from '@ember/object';
+import { task, timeout } from 'ember-concurrency';
+
+const secondsMap = {
+  s: 1,
+  m: 60,
+  h: 3600,
+  d: 86400,
+};
+const convertToSeconds = (time, unit) => {
+  return time * secondsMap[unit];
+};
+const convertFromSeconds = (seconds, unit) => {
+  return seconds / secondsMap[unit];
+};
+
+export default Component.extend({
+  enableTTL: false,
+  label: 'Time to live (TTL)',
+  helperTextDisabled: 'Allow tokens to be used indefinitely',
+  helperTextEnabled: 'Disable the use of the token after',
+  time: 30,
+  unit: 'm',
+  unitOptions: computed(function() {
+    return [
+      { label: 'seconds', value: 's' },
+      { label: 'minutes', value: 'm' },
+      { label: 'hours', value: 'h' },
+      { label: 'days', value: 'd' },
+    ];
+  }),
+
+  onChange: ttl => {
+    console.log({ ttl });
+  },
+
+  TTL: computed('enableTTL', 'seconds', function() {
+    let { time, unit, enableTTL, seconds } = this.getProperties('time', 'unit', 'enableTTL', 'seconds');
+    return {
+      enabled: enableTTL,
+      seconds,
+      timeString: time + unit,
+    };
+  }),
+
+  updateTime: task(function*(newTime) {
+    this.set('time', newTime);
+    this.onChange(this.TTL);
+    if (Ember.testing) {
+      return;
+    }
+    this.set('recalculateSeconds', true);
+    yield timeout(5000);
```

Comment:
will 5 seconds always be the timeout used, or can you imagine a case where it might need to change?  I honestly don't know but thought it worth asking.  If yes, then again, maybe pass in as a prop, and have 5 seconds be the default?

Reply:
That's a great question, and it's worth just saving as a default prop and allowing it to be overwritten, even if in the short term it will be 5 seconds